internal/engine/build_and_deployer_test.go

```diff
@@ -424,3 +411,11 @@ func (f *bdFixture) withContainerForBuild(build BuildResult) *bdFixture {
 	f.bd.PostProcessBuild(f.ctx, build)
 	return f
 }
+
+func (f *bdFixture) assertContainerRestarts(count int) {
+	expected := map[string]int{}
```

Comment:
...i'm confused, why not just `assert.Equal(f.T(), count, f.docker.RestartsByContainer[build.MagicTestContainerID]`?

Reply:
i'm trying to avoid the bug where we assert 0 restarts, but some other container id got restarted

Reply:
aah. good catch! (Also, if i came upon this code, i would probably just change it... maybe leave a comment?)

another possibility:
```
if assert.Equal(f.T(), 1, len(f.docker.RestartsByContainer, "these tests should only ever effect a single container") {
    assert.Equal(f.T(), count, f.docker.RestartsByContainer[build.MagicTestContainerID])
}
```
?